joomla/cassiopeia
# [Google Fonts] Local fonts enhancements

Pull Request for PR #18 .

### Summary of Changes

- Move the fonts abd their scss to `build/media_source/fonts`
- Modify the the css build script so it copies the fonts and the compiled css to `media/fonts`.
- Use a `filelist` field for selecting the fonts => No need to edit `templateDetails.xml` anymore for adding new fonts.
- Dynamically register the asset so no need to edit `joomla.asset.json` anymore for adding new fonts.

### Testing Instructions

1. Checkout the branch of this PR and if necessary pull changes.
2. Run `npm ci`.
3. In backend in Cassiopeia's template style's options, switch on Google Fonts.
4. Select one of the 2 available fonts and save.
5. Check in frontend and with your browser's developer tools that the selected font is used.
6. In backend, select the other font and save.
7. Check in frontend and with your browser's developer tools that the selected font is used.
8. In backend switch off Google Fonts.
9. Check in frontend and with your browser's developer tools that no local font is used.

### Expected result

No need to edit any files for adding new fonts after installation. You can just upload the fonts and the css (minified or not, or both) to the `media/fonts` folder, and they will be available.

Any template can use the local fonts.

### Actual result

You have to edit `templateDetails.xml` and `joomla.asset.json` for adding new fonts after installation.

Only Cassiopeia can use the local fonts.

### Maybe to be improved with other PR(s)

Currently it will not work if the fonts and their css are located in different sub-folders of the `media/fonts` folder for each font.

Maybe the folder where to look for fonts could be configurable in backend in the template style options.

Maybe the one or 2 fonts we will ship could be fetched with a little js by the css build script, or we could use node and some external node package to maintain the dependency and fetch them if necessary, so we not have to maintain them in the media sources.

### Documentation Changes Required

None in addition to what's required for PR #18 .
